xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1159,9 +1160,17 @@ bool CPVRClients::UpdateAndInitialiseClients(bool bInitialiseAllClients /* = fal
     }
   }
 
+  // indicate that addons have now been initialised
+  m_loadedEvent.Set();
+
   return bReturn;
 }
 
+void CPVRClients::WaitForInitialisation()
+{
+  m_loadedEvent.Wait();
```

Comment:
shouldn't wait for all add-ons to be loaded forever anyway. and you need to signal the event when the pvrmanager is stopped, or this will deadlock